src/Domain/ValueObjects/ShoppingItemFilterEnum.php

```diff
@@ -2,29 +2,46 @@
 
 namespace Lindyhopchris\ShoppingList\Domain\ValueObjects;
 
+use http\Exception\InvalidArgumentException;
+
 class ShoppingItemFilterEnum
 {
+
+    const ONE = 'all';
+    const TWO = 'only completed';
+    const THREE = 'only not completed';
+
+    function showConstant()
+    {
+        echo self::ONE;
+        echo self::TWO;
+        echo self::THREE;
+    }
     /**
      * @var string
      */
-    private string $all;
+    private string $value;
 
     /**
-     * @param string $all
+     * @param string $value
      */
-    public function __construct(string $all)
+    public function __construct(string $value)
     {
-        $this->all = $all;
+        if ($value !== self::ONE || self::TWO || self::THREE) {
```

Comment:
Required: close, but this doesn't do what you're trying to do. Don't worry about it now though - we will "discover" that this doesn't work when we write the tests.